### PR TITLE
Upgrade to clang-tidy and clang-format to version 8.

### PR DESCRIPTION
**Pull request by patrickvacek**
Debian testing no longer supports v6 so our hand has been forced. Fixes https://github.com/advancedtelematic/aktualizr/issues/1675.

### CODE REVIEW

**Comment by lbonn** [comment]
You're confident the tests will pass? I was about to open https://github.com/advancedtelematic/aktualizr/pull/1678 to buy us some time.

**Comment by patrickvacek** [comment]
> You're confident the tests will pass?

Nope, not at all. I was trying to test locally with docker but ran into some problems, so figured I'd enlist CI to speed up the process and share my progress. If this fails and https://github.com/advancedtelematic/aktualizr/pull/1678 succeeds I have no problem with merging that first just to get things running again.

File: src/libaktualizr/crypto/crypto.cc [hunk 1]
```diff
@@ -95,6 +95,7 @@ Json::Value PublicKey::ToUptane() const {
 
 std::string PublicKey::KeyId() const {
   std::string key_content = value_;
+  // NOLINTNEXTLINE(clang-analyzer-cplusplus.NewDeleteLeaks)
```
Comment (lbonn) [inline src/libaktualizr/crypto/crypto.cc:98]: nice!

**Comment by patrickvacek** [comment]
I'm not so sure this works as I was hoping. This doesn't look good to me:
```
+ cmake --build . --target clang-tidy -- -j6 -k 0
ninja: no work to do.
```

**Comment by lbonn** [comment]
> I'm not so sure this works as I was hoping. This doesn't look good to me:
> 
> ```
> + cmake --build . --target clang-tidy -- -j6 -k 0
> ninja: no work to do.
> ```

That's what I was afraid of, this file doesn't necessarily exist or is up to date at configure time.

Another hacky way which could work better is to add this check in the `COMMAND` argument of the `add_custom_target`, later down.

**Comment by patrickvacek** [comment]
> Another hacky way which could work better is to add this check in the `COMMAND` argument of the `add_custom_target`, later down.

I tried that, but it's not so easy. If I do `grep ... || clang-tidy ...` then the command fails if grep fails. If I wrap that all in `|| true` but then it wouldn't fail if clang-tidy fails. I wish I could just make the target depend on `compile_commands.json` existing, but I'm not aware of an easy way to do that.

**Comment by patrickvacek** [comment]
It appears to be working now, but for some reason it's dumping a bunch of output that wasn't there before, despite that the `-quiet` is still there. I can't figure that one out.

**Comment by lbonn** [comment]
> > Another hacky way which could work better is to add this check in the `COMMAND` argument of the `add_custom_target`, later down.
> 
> I tried that, but it's not so easy. If I do `grep ... || clang-tidy ...` then the command fails if grep fails. If I wrap that all in `|| true` but then it wouldn't fail if clang-tidy fails. I wish I could just make the target depend on `compile_commands.json` existing, but I'm not aware of an easy way to do that.

What about?

```
! grep ... || clang-tidy ...
```

**Comment by patrickvacek** [comment]
> What about?
> 
> ```
> ! grep ... || clang-tidy ...
> ```

Ah yes, that's what I was looking for. Well, the script is more powerful and easier to use anyway. And it has a check to make sure it's actually finding `compile_command.json` and not just skipping everything!

File: scripts/clang-tidy-wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+FILE="${4}"
+
+if [[ ! -e "${CMAKE_BINARY_DIR}/compile_commands.json" ]]; then
+  echo "compile_commands.json not found!"
+  exit 1
+fi
+
+# Filter out anything that we aren't trying to compile. Older clang-tidy
+# versions (<=6) skipped them automatically.
+if grep "${FILE}" "${CMAKE_BINARY_DIR}/compile_commands.json"; then
```
Comment (lbonn) [inline scripts/clang-tidy-wrapper.sh:17]:
```suggestion
if grep -q "${FILE}" "${CMAKE_BINARY_DIR}/compile_commands.json"; then
```

Comment (lbonn) [reply scripts/clang-tidy-wrapper.sh:17]: Oops 2 seconds too late :)

Comment (patrickvacek) [reply scripts/clang-tidy-wrapper.sh:17]: Sorry! Now I see what's going on. Wanna make a PR for that or shall I?

Comment (lbonn) [reply scripts/clang-tidy-wrapper.sh:17]: Ok I'll do it quick

**Pull request merged**